modules/common/src/test/scala/surge/internal/streams/StreamManagerSpec.scala

```diff
@@ -218,6 +218,33 @@ class StreamManagerSpec
       }
     }
 
+    "Be able to replay an empty stream" in {
+      withRunningKafkaOnFoundPort(EmbeddedKafkaConfig(kafkaPort = 0, zooKeeperPort = 0)) { implicit actualConfig =>
+        val topic = KafkaTopic("testTopic4")
+        createCustomTopic(topic.name, partitions = 3)
+        val embeddedBroker = s"localhost:${actualConfig.kafkaPort}"
+        val probe = TestProbe()
+
+        val settings = KafkaForeverReplaySettings(defaultConfig, topic.name).copy(brokers = List(embeddedBroker))
+        val completeProbe = TestProbe()
+        val kafkaForeverReplayStrategy =
+          new KafkaForeverReplayStrategyImpl(defaultConfig, system, settings, () => Future.successful(true), () => completeProbe.ref ! ReplayComplete)
+        val consumer =
+          testStreamManager(topic, kafkaBrokers = embeddedBroker, groupId = "replay-test", sendToTestProbe(probe), kafkaForeverReplayStrategy, settings)
+
+        consumer.start()
+
+        // fix: if we attempt to replay prior to a stream manager being registered for the topic replay will fail.
+        //  we need a to ensure the streamManager is started prior to calling replay.
+        Thread.sleep(5.seconds.toMillis)
```

Comment:
Good for now but we'll want to circle back and make this more explicit - we could send it an "Are you ready" type message and assert we get a response back rather than the sleep here.